homeassistant/components/sensor/__init__.py

```diff
@@ -95,21 +96,43 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return await component.async_unload_entry(entry)
 
 
+@dataclass
+class SensorEntityDescription(EntityDescription):
+    """An class that describes sensor entities."""
+
+    state_class: str | None = None
+    last_reset: datetime | None = None
+
+
 class SensorEntity(Entity):
     """Base class for sensor entities."""
 
-    _attr_state_class: str | None = None
-    _attr_last_reset: datetime | None = None
+    _attr_entity_description: SensorEntityDescription | None = None
```

Comment:
This would then be something like this
```py
    _attr_state_class: str | None = None  # Preserve defaults in case method isn't called
    _attr_last_reset: datetime | None = None

    def _set_attributes_from_description(desc: SensorEnityDescription) -> None:
        self._attr_state_class = desc.state_class
        self._attr_last_reset = desc.last_reset
        super()._set_attributes_from_description(desc)
```